core/src/main/java/hivemall/lda/OnlineLDAModel.java

```diff
@@ -0,0 +1,464 @@
+
+    // positive value which downweights early iterations
+    private double tau0_ = 1020;
+
+    // exponential decay rate (i.e., learning rate) which must be in (0.5, 1] to guarantee convergence
+    private double kappa_ = 0.7;
+
+    // random number generator
+    GammaDistribution gd;
+    private double SHAPE = 100d;
+    private double SCALE = 1.d / SHAPE;
+
+    // parameters
+    private List<Map<String, float[]>> phi_;
+    private float[][] gamma_;
+    private Map<String, float[]> lambda_;
+
+    // check convergence in the expectation (E) step
+    private double DELTA = 1E-5;
+
+    // for update size of lambda_
+    private int dummySize = 100;
+    private float[][] dummyLambdas;
+
+    private List<Map<String, Float>> miniBatchMap;
+    private int miniBatchSize;
+
+    private int accumDocCount = 0;
+    private int accumWordCount = 0;
+
+    public OnlineLDAModel(int K, float alpha, float eta, int D, double tau0, double kappa) {
+        Preconditions.checkArgument(0.d < tau0, "tau0 MUST be positive: " + tau0);
+        Preconditions.checkArgument(0.5 < kappa && kappa <= 1.d,
+            "kappa MUST be in (0.5, 1.0]: " + kappa);
+
+        K_ = K;
+        alpha_ = alpha;
+        eta_ = eta;
+        D_ = D;
+        tau0_ = tau0;
+        kappa_ = kappa;
+
+        // initialize a random number generator
+        gd = new GammaDistribution(SHAPE, SCALE);
+        gd.reseedRandomGenerator(1001);
+
+        // initialize the parameters
+        lambda_ = new HashMap<String, float[]>();
+
+        setDummyLambda();
+    }
+
+    private void setDummyLambda() {
+        dummyLambdas = new float[dummySize][];
+        double[] tmpDArray;
+        for (int b = 0; b < dummySize; b++) {
+            float[] tmpDummyLambda = new float[K_];
+            tmpDArray = gd.sample(K_);
+            for (int k = 0; k < K_; k++) {
+                tmpDummyLambda[k] = (float) tmpDArray[k];
+            }
+            dummyLambdas[b] = tmpDummyLambda;
+        }
+    }
+
+    public void train(String[][] miniBatch, int time) {
+        miniBatchSize = miniBatch.length;
+
+        rhot = Math.pow(tau0_ + time, -kappa_);
+
+        // get the number of words(Nd) for each documents
+        getMiniBatchParams(miniBatch);
+        accumDocCount += miniBatchSize;
+
+        makeMiniBatchMap(miniBatch);
+
+        updateSizeOfParameterForMiniBatch();
+
+        // Expectation
+        float[][] lastGamma;
+        do {
+            // (deep) copy the last gamma values
+            lastGamma = new float[gamma_.length][];
+            for (int d = 0; d < gamma_.length; d++) {
+                lastGamma[d] = gamma_[d].clone();
+            }
+
+            stepE();
+        } while (!checkGammaDiff(lastGamma, gamma_));
+
+        // Maximization
+        stepM();
+    }
+
+    private void stepE() {
+        // 1) Updating phi_
+
+        // dirichlet_expectation_2d(gamma_)
+        float[][] ElogTheta = new float[miniBatchSize][K_];
```

Comment:
It's true, but plz wait until the final push after WIP condition. I will later check if the algorithm is implemented correctly, and a new variable e.g. `expElogTheta` might be introduced. If so, I guess `ElogTheta` is better in terms of both readability. It should be easy to understand a relationship between `ElogTheta` and its exponential.